specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json

```diff
@@ -328,46 +336,521 @@
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update a registered ASN for the peering": {
+            "$ref": "./examples/CreateRegisteredAsn.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "PeeringRegisteredAsns"
+        ],
+        "description": "Deletes an existing registered ASN with the specified name under the given subscription, resource group and peering.",
+        "operationId": "RegisteredAsns_Delete",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of the resource group.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "peeringName",
+            "in": "path",
+            "description": "The name of the peering.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "registeredAsnName",
+            "in": "path",
+            "description": "The name of the registered ASN.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "NoContent"
+          },
+          "default": {
+            "description": "Error response describing why the operation has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Deletes a registered ASN associated with the peering": {
+            "$ref": "./examples/DeleteRegisteredAsn.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns": {
+      "get": {
+        "tags": [
+          "PeeringRegisteredAsns"
+        ],
+        "description": "Lists all registered ASNs under the given subscription, resource group and peering.",
+        "operationId": "RegisteredAsns_ListByPeering",
```

Comment:
>RegisteredAsns_ListByPeering [](start = 24, length = 28)

also add _ListBySubscription and _ListByResourceGroup

Reply:
This is a child resource of Peering. Do we still need to add this?